### PR TITLE
Remove listing of undefined TPL deps (#11152)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
This removes a bunch of  undefined TPL deps.  For details, see [tribits/CHANGELOG.md entry from 2022-10-11](https://github.com/TriBITSPub/TriBITS/blob/515df6c76246b28b8bef1cd637622eda0bb49e9f/tribits/CHANGELOG.md#2022-10-11) from updated TriBITS 'master' from https://github.com/TriBITSPub/TriBITS/pull/530.

This is needed to fix Charon Trilinos integration which uses TriBITS 'master' (see #11152).

I posted the matching external repo PRs in:

* https://github.com/kokkos/kokkos/pull/5568
* https://github.com/kokkos/kokkos-kernels/pull/1568
* https://github.com/sandialabs/seacas/pull/334

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
Can someone please approve ASAP?  Getting this merged will fix Charon Trilinos integration.

NOTE: this works with both old and new TriBITS so this PR should be merged now even before the update of TriBITS snapshot into Trilinos 'develop'.

**Comment by bartlettroscoe** [comment]
FYI: I posted the matching external repo PRs in:

* https://github.com/kokkos/kokkos/pull/5568
* https://github.com/kokkos/kokkos-kernels/pull/1568
* https://github.com/sandialabs/seacas/pull/334

**Review by ndellingwood — APPROVED** [review]
Approving based on kokkos and kokkos-kernels changes (PRs with matching changes to the respective repos have been approved as well)

@bartlettroscoe I'd prefer to hold off the merge of this PR until https://github.com/kokkos/kokkos-kernels/pull/1568 completes CI if that is okay (the changes should have no impact on standalone kokkos-kernels testing, but I'll feel better once the testing completes)

**Comment by ndellingwood** [comment]
PR https://github.com/kokkos/kokkos-kernels/pull/1568 merged into kokkos-kernels, thanks @bartlettroscoe !

File: packages/seacas/libraries/ioss/cmake/Dependencies.cmake
Comment (bartlettroscoe) [inline packages/seacas/libraries/ioss/cmake/Dependencies.cmake:9]: FYI: After the merge of the related PR #11156, the next snapshot of SEACS will pull in:

https://github.com/sandialabs/seacas/blob/master/packages/seacas/libraries/ioss/cmake/Dependencies.cmake

which has:

```
TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
  LIB_OPTIONAL_PACKAGES SEACASExodus Pamgen Zoltan Kokkos
  LIB_OPTIONAL_TPLS HDF5 CGNS ParMETIS Faodel Cereal DLlib Pthread DataWarp ADIOS2 Catalyst2 ${SEACAS_GTest_TPL_name}
)
```

That will make it so that this file will not list the `GTest` TPL when configuring SEACAS under Trilinos.

But for now, the changes in this particular PR are a good short-term solution (because removing these listed TPLs is what was happening anyway).

**Review by bartlettroscoe — COMMENTED** [review]
This is okay to merge.  See my comment in https://github.com/trilinos/Trilinos/pull/11154/files#r998765365.

**Pull request merged**
